.github/workflows/ci.yml

```diff
@@ -4,41 +4,95 @@ on: [push, pull_request]
+            python-version: '3.9'
+    env:
+      OS: ${{ matrix.os }}
+      PYTHON: ${{ matrix.python-version }}
+
+    name: ${{ matrix.os }} - Python ${{ matrix.python-version }}
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python }}
-      uses: actions/setup-python@v1
+
+    - name: Acquire sources
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
       with:
-        python-version: ${{ matrix.python }}
-    - name: Install APT packages
-      if: contains(${{ matrix.os }}, 'ubuntu')
-      run: |
-        sudo apt install libbz2-dev libhdf5-serial-dev liblzo2-dev
-    - name: Install dependencies
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install Debian packages
+      if: contains(matrix.os, 'ubuntu')
       run: |
-        python -m pip install --upgrade setuptools pip wheel
-        python -m pip install -r requirements.txt
-        python -m pip install cython>=0.29.21
-        # python -m pip install sphinx>=1.1 sphinx_rtd_theme numpydoc ipython
-    - name: Source distribution
+        sudo apt install libblosc-dev libbz2-dev libhdf5-dev liblz4-dev liblzo2-dev libsnappy-dev libzstd-dev zlib1g-dev
+
+    - name: Install Homebrew packages
+      if: contains(matrix.os, 'macos')
+      env:
+        # Set brew cleanup period to 10 years, essentially disabling cleanup.
+        HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS: 3650
       run: |
-        python setup.py sdist
-    - name: Installation
+        brew install c-blosc bzip2 hdf5 lz4 lzo snappy zstd zlib
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      env:
+        # Increase this value to reset cache if etc/example-environment.yml has not changed
+        CACHE_NUMBER: 0
+      with:
+        path: |
+          ~/.cache/pip
+          ~/conda_pkgs_dir
+        key:
+          ${{ runner.os }}-pythonpkg-${{ env.CACHE_NUMBER }}-${{ hashFiles('requirements.txt', 'requirements-docs.txt') }}
+
+    - name: Install dependencies on Linux and macOS
+      if: contains(matrix.os, 'windows') == false
+      run: pip install --requirement=requirements.txt --requirement=requirements-dev.txt
+
+    - name: Install Miniconda on Windows
+      if: contains(matrix.os, 'windows')
+
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies on Windows
+      if: contains(matrix.os, 'windows')
       run: |
-        pip install dist/*.tar.gz
-    - name: 'Run test'
+
+        # Install packages.
+        conda create --yes python=${{ matrix.python-version }}
+        conda install --yes blosc bzip2 hdf5 lz4 lzo snappy zstd zlib
+
+        # Install build dependencies.
+        conda install --yes --file=requirements.txt --file=requirements-dev.txt
+
+    - name: Run build and install in development mode
+      run: python setup.py develop
```

Comment:
Is there a particular reason to build/install in developer mode?
I would prefer to stay with the old approach to ``pip install`` from the source tarball.
This allows to ensure the the sdist is properly working and using pip to install will be, let's say, required when we will migrate to a more modern setup environment (with pyproject.toml).